app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -167,6 +169,18 @@ def get_managed_images
         process_collection(images, :vms) { |image| parse_managed_image(image) }
       end
 
+      def get_market_images
+        urns   = @options.market_image_urns
+        images = if urns
+                   gather_data_for_this_region(@vmis).select do |image|
+                     urns.include?(image.id)
+                   end
+                 else
+                   gather_data_for_this_region(@vmis)
+                 end
```

Comment:
`az vm image list --output json --all` seems to be considerably faster than our custom method. I'm not sure why yet, still digging through the output (with --debug attached to the command).

Maybe we can speed things up and/or just cache the results.